File: ios/Runner/CaddyMethodHandler.swift
import Flutter
import UIKit

class CaddyMethodHandler: NSObject {
    private let channel: FlutterMethodChannel

    init(messenger: FlutterBinaryMessenger) {
        channel = FlutterMethodChannel(name: "com.caddy_app/caddy", binaryMessenger: messenger)
        super.init()
        channel.setMethodCallHandler(handle)
    }

    private func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        DispatchQueue.global(qos: .userInitiated).async {
            switch call.method {
            case "start":
                let args = call.arguments as? [String: Any]
                let config = args?["config"] as? String ?? "{}"
                let response = CaddyBridgeStartCaddy(config)
                DispatchQueue.main.async { result(response) }
            case "stop":
                let response = CaddyBridgeStopCaddy()
                DispatchQueue.main.async { result(response) }
            case "reload":
                let args = call.arguments as? [String: Any]
                let config = args?["config"] as? String ?? "{}"
                let response = CaddyBridgeReloadCaddy(config)
                DispatchQueue.main.async { result(response) }
            case "status":
                let response = CaddyBridgeGetCaddyStatus()
                DispatchQueue.main.async { result(response) }
            case "setEnvironment":
                let args = call.arguments as? [String: Any]
                let env = args?["env"] as? String ?? "{}"
                let response = CaddyBridgeSetEnvironment(env)
                DispatchQueue.main.async { result(response) }
            default:
                DispatchQueue.main.async { result(FlutterMethodNotImplemented) }
            }
        }
    }

    func dispose() {
        channel.setMethodCallHandler(nil)
    }
}

File: android/app/src/main/kotlin/app/gsmlg/caddyapp/CaddyMethodHandler.kt
package app.gsmlg.caddyapp

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class CaddyMethodHandler(messenger: BinaryMessenger) :
    MethodChannel.MethodCallHandler {

    private val channel = MethodChannel(messenger, "com.caddy_app/caddy")
    private val scope = CoroutineScope(Dispatchers.IO)

    init {
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        scope.launch {
            try {
                val response = when (call.method) {
                    "start" -> {
                        val config = call.argument<String>("config") ?: "{}"
                        caddy_bridge.CaddyBridge.startCaddy(config)
                    }
                    "stop" -> {
                        caddy_bridge.CaddyBridge.stopCaddy()
                    }
                    "reload" -> {
                        val config = call.argument<String>("config") ?: "{}"
                        caddy_bridge.CaddyBridge.reloadCaddy(config)
                    }
                    "status" -> {
                        caddy_bridge.CaddyBridge.getCaddyStatus()
                    }
                    else -> {
                        result.notImplemented()
                        return@launch
                    }
                }
                result.success(response)
            } catch (e: Exception) {
                result.error("CADDY_ERROR", e.message, null)
            }
        }
    }

    fun dispose() {
        channel.setMethodCallHandler(null)
    }
}
